sdk/src/main/java/com/mongodb/stitch/android/auth/oauth2/facebook/FacebookAuthProviderInfo.java

```diff
@@ -1,42 +1,96 @@
 package com.mongodb.stitch.android.auth.oauth2.facebook;
 
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.mongodb.stitch.android.auth.AuthProviderInfo;
 
 import java.util.List;
 
 /**
  * FacebookAuthProviderInfo contains information needed to create a {@link FacebookAuthProvider}.
  */
-public class FacebookAuthProviderInfo {
-    public static final String FQ_NAME = "oauth2/facebook";
+public class FacebookAuthProviderInfo extends AuthProviderInfo {
+    public static final String FQ_NAME = "oauth2-facebook";
 
-    private final String _appId;
-    private final List<String> _scopes;
+    static final String METADATA_FIELDS = "metadata_fields";
 
-    @JsonCreator
-    private FacebookAuthProviderInfo(
+    public static class MetadataField {
+        class Fields {
+            static final String NAME = "name";
+            static final String REQUIRED = "required";
+        }
+
+        private String _name;
+        private boolean _required;
+
+        private MetadataField(@JsonProperty(Fields.NAME)
+                              @NonNull String name,
+                              @JsonProperty(Fields.REQUIRED)
+                                      boolean required) {
+            this._name = name;
+            this._required = required;
+        }
+
+        @JsonProperty(Fields.NAME)
+        @NonNull
+        public String getName() {
+            return this._name;
+        }
+
+        @JsonProperty(Fields.REQUIRED)
+        @NonNull
+        public boolean getRequired() {
+            return this._required;
+        }
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class Config {
+        static class Fields {
+            static final String CLIENT_ID = "clientId";
```

Comment:
Should this be "client_id"?

Reply:
Unfortunately we didn't update these to snakecase on the serverside